### PR TITLE
New item tiles

### PR DESCRIPTION
**Pull request by bhollis**
This is a very rough work in progress branch for a second stab at new item tiles. The majority of it is in `DarkTile.tsx` which is a bad name for it.

screen shot 2018-10-08 at 6 37 35 pm

@SunburnedGoose if you wanna take a crack at this feel free!

### CODE REVIEW

**Comment by SunburnedGoose** [comment]
The issue I've been working on here is the Masterwork graphics. I need to drop the grey border and replace it with the gold border/graphic of the MW icon.  When I do that, the grey box below it looks out of place.

**Comment by bhollis** [comment]
I’ve been thinking that maybe we want to edit the masterwork graphics or stick with our own indicators. I don’t think the in-game ones look very good.

**Comment by bhollis** [comment]
Before:
screen shot 2018-11-11 at 12 46 28 am

After:
screen shot 2018-11-11 at 12 46 52 am

I think this is good to go to beta at least. A couple things were changed that are *not* under a feature flag:

1. Switched the "favorite" tag to a heart instead of a star, so it doesn't overlap with the ratings star.
2. Remove count for stacks of one.

Everything else is only on when the feature flag is on:

1. Taller tiles that show the whole item picture.
2. Element icon next to power.
2. Lock icon for locked items.
3. Tag/lock now in the bottom left, and are green. They're green because it's the only color that'll stand out against yellow, purple, and blue.
4. Ratings are smaller and have a star icon, and only get colored if it's a perfect 5.
5. Most sizing is proportional to the tile size, so adjusting the size (including in mobile) should work better.
5. Use the in-game highlighting for masterworks, but inset it so it doesn't go over our border.
6. Moved the progress bar for objective-laden items to the top, and made it thicker.

screen shot 2018-11-11 at 12 51 49 am

I tried with and without the ammo indicator, and decided on not, since people can memorize what weapons take what ammo pretty easily.

D1 requires a little polish but isn't bad:

screen shot 2018-11-11 at 12 54 06 am

There's also sort of a bug where any tile that doesn't have a label is too tall, and the move popup kind of attaches to nowhere. I can fix that up pretty easily later.

**Comment by bhollis** [comment]
screen shot 2018-11-11 at 1 03 36 am

I'm tempted to space them out a bit more...

**Comment by bhollis** [comment]
I'll definitely have to try it - the 1px border looks better now than the last time I tried it, especially if I can use that to get a bit more space between items.

Making reviews small is definitely odd, but rounding is even weirder. There's just still a lot of difference between 4.0 and 5.0. And if I make them full size they but right up against the element. I also really like having the element in the tag.

The lime green icons take some getting used to, but at least they stand out! I was also thinking about having a gradient behind them, and I can try that too.

I don't think having lock override tags is great, because they sorta mean different things. I also tried the experiment of replacing the ratings star with the tag, but it gets confusing then to explain what the number is for.

**Comment by bhollis** [comment]
screen shot 2018-11-11 at 4 51 50 pm

Updated for thinner borders and more space between tiles, plus some other tweaks.

**Comment by bhollis** [comment]
BTW, looking at it, any darker and contrast suffers.

File: src/app/inventory/TallTile.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  badgeInfo: BadgeInfo;
+  rating?: number;
+  hideRating?: boolean;
+  tag?: TagValue;
+  isNew: boolean;
+}) {
+  const borderless =
+    (item.isDestiny2 &&
+      item.isDestiny2() &&
+      (item.bucket.hash === 3284755031 || item.itemCategoryHashes.includes(268598612))) ||
```
Comment (kyleshay) [inline src/app/inventory/TallTile.tsx]: this line is throwing an error on the settings page (itemCategoryHashes is undefined on our fake item display object.)

**Comment by duckmanBR** [comment]
Other than the taller tiles, eagerly waiting for this to be merged on beta...

**Pull request merged**